### PR TITLE
[Dashboard] Support ray status output to the dashboard job page

### PR DESCRIPTION
**Pull request by rkooo567**
Signed-off-by: SangBin Cho <rkooo567@gmail.com>

## Why are these changes needed?

Screen Shot 2023-01-29 at 4 38 54 PM

This is the initial prototype of integrating ray status to the frontend.

I think we could've returned structured data from the backend, but I decided to parse ray status output from the frontend for quick implementation (so that we can support if from ray 2.3). 

Screen Shot 2023-01-29 at 4 38 54 PM

here's the initial UI for the prototype. Please let me know what's the final state we should have for 2.3 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by scottsun94** [comment]
Here is my proposal.
- make sure it is scrollable within the card in case the output is long (p0)
- visual polish - don't leave so much white space. See the 2nd screenshot for the spec (p0)
- separate the output into 2 cards (p1) and also include the node count (p1)
- mentioned in another PR, we should put these cards into expandable sections. And we probably should only expand task progress bar by default to improve the performance?

Screen Shot 2023-01-29 at 9 11 39 PM
Screen Shot 2023-01-29 at 9 20 50 PM

**Comment by rkooo567** [comment]
@scottsun94 

> make sure it is scrollable within the card in case the output is long (p0)

This is already done

> visual polish - don't leave so much white space. See the 2nd screenshot for the spec (p0)

I will ask @alanwguo in person to figure it out

For other P1s, I wouldn't handle them in this PR

**Comment by rkooo567** [comment]
Screen Shot 2023-01-31 at 2 42 26 AM

Hmm this is the best I could do.. I may need help for those correct spacing

File: dashboard/client/src/pages/job/JobDetail.tsx [hunk 1]
```diff
@@ -35,6 +36,62 @@ export const JobDetailChartsPage = ({
...
+  console.log(cluster_status?.data.clusterStatus.split("\n"));
+
+  const FormatNodeStatus = (cluster_status: string) => {
+    // ==== auto scaling status
+    // Node status
+    // ....
+    // Resources
+    // ....
+    const sections = cluster_status.split("Resources");
+    return FormatClusterStatus(
```
Comment (alanwguo) [inline dashboard/client/src/pages/job/JobDetail.tsx:49]:
```suggestion
    return formatClusterStatus(
```

File: dashboard/client/src/pages/job/JobDetail.tsx [hunk 2]
```diff
@@ -35,6 +36,62 @@ export const JobDetailChartsPage = ({
...
+      "Node Status",
+      sections[0].split("Node status")[1],
+    );
+  };
+
+  const FormatResourcesStatus = (cluster_status: string) => {
```
Comment (alanwguo) [inline dashboard/client/src/pages/job/JobDetail.tsx:55]:
```suggestion
  const formatResourcesStatus = (cluster_status: string) => {
```

File: dashboard/client/src/pages/job/JobDetail.tsx [hunk 3]
```diff
@@ -35,6 +36,62 @@ export const JobDetailChartsPage = ({
...
+    // ==== auto scaling status
+    // Node status
+    // ....
+    // Resources
+    // ....
+    const sections = cluster_status.split("Resources");
+    return FormatClusterStatus("Resource Status", sections[1]);
```
Comment (alanwguo) [inline dashboard/client/src/pages/job/JobDetail.tsx:62]:
```suggestion
    return formatClusterStatus("Resource Status", sections[1]);
```

File: dashboard/client/src/pages/job/JobDetail.tsx [hunk 4]
```diff
@@ -35,6 +36,62 @@ export const JobDetailChartsPage = ({
...
+  };
+
+  const FormatClusterStatus = (title: string, cluster_status: string) => {
```
Comment (alanwguo) [inline dashboard/client/src/pages/job/JobDetail.tsx:65]:
```suggestion
  const formatClusterStatus = (title: string, cluster_status: string) => {
```

File: dashboard/client/src/pages/job/JobDetail.tsx [hunk 5]
```diff
@@ -126,6 +183,46 @@ export const JobDetailChartsPage = ({
...
+              flexDirection="column"
+              height="300px"
+              style={{
+                overflow: "hidden",
+                overflowY: "scroll",
+              }}
+              sx={{ borderRadius: "16px" }}
+            >
+              {cluster_status?.data
+                ? FormatNodeStatus(cluster_status?.data.clusterStatus)
```
Comment (alanwguo) [inline dashboard/client/src/pages/job/JobDetail.tsx:201]:
```suggestion
                ? formatNodeStatus(cluster_status?.data.clusterStatus)
```

File: dashboard/client/src/pages/job/JobDetail.tsx [hunk 6]
```diff
@@ -126,6 +183,46 @@ export const JobDetailChartsPage = ({
...
+              flexDirection="column"
+              height="300px"
+              style={{
+                overflow: "hidden",
+                overflowY: "scroll",
+              }}
+              sx={{ border: 1, borderRadius: "1", borderColor: "primary.main" }}
+            >
+              {cluster_status?.data
+                ? FormatResourcesStatus(cluster_status?.data.clusterStatus)
```
Comment (alanwguo) [inline dashboard/client/src/pages/job/JobDetail.tsx:220]:
```suggestion
                ? formatResourcesStatus(cluster_status?.data.clusterStatus)
```

File: dashboard/client/src/pages/job/JobDetail.tsx (hunk 3 above)
Comment (rkooo567) [reply dashboard/client/src/pages/job/JobDetail.tsx:62]: so inlined component should start with lowercase?

**Review by alanwguo — APPROVED** [review]

**Pull request merged**